### ISSUE TITLE
Background notifications only updates other person's message when chatting in web with iOS push notification turned ON

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16258]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login as user A on both web and iOS
2. Chat with user B on web
3. Continue chatting with several messages between user A and B
4. Tap on the notification in iOS 

## Expected Result:
Able to see both user A and user B chats in the report

## Actual Result:
 When opening chat on iOS able to see only user B' messages, and not user A's before the API request finished loading

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.87-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/226388224-680e448d-c4ba-472e-bc4b-f64e10cf2075.MP4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @AndrewGable 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1679161604171819

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~011683f05700e17640</li>
        <li>Upwork Job ID: 1637855294320246784</li>
        <li>Last Price Increase: 2023-03-20</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by Julesssss** [comment]
Adding this to our notification tracker. This is going to be resolved by our wider plans for improved notification delivery. Here's a good [overview](https://github.com/Expensify/Expensify/issues/255265#issuecomment-1476568767).

**Comment by Julesssss** [comment]
Taking the issue as it will be resolved by our planned changes detailed [here](https://github.com/Expensify/Expensify/issues/255265#issuecomment-1476568767).

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~011683f05700e17640

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor Plus for review of internal employee PR - @eVoloshchak (`Internal`)

**Comment by kadiealexander** [comment]
Not overdue, being worked on here https://github.com/Expensify/Expensify/issues/255265#issuecomment-1476568767.

**Comment by kadiealexander** [comment]
Placing on hold and shifting to weekly while the team work through [this project](https://github.com/Expensify/Expensify/issues/255265#issuecomment-1476568767).

**Comment by Julesssss** [comment]
I'll work on this next. SHould be unblocked now 👍

**Comment by MelvinBot** [comment]
@eVoloshchak @Julesssss @kadiealexander this issue was created 2 weeks ago. Are we close to a solution? Let's make sure we're treating this as a top priority. Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by MelvinBot** [comment]
@eVoloshchak @Julesssss @kadiealexander this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a>. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by MelvinBot** [comment]
@eVoloshchak @Julesssss @kadiealexander this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a>. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by Julesssss** [comment]
No need for this to be daily. We'll be continuing to make progress shortly.

**Comment by MelvinBot** [comment]
@eVoloshchak @Julesssss @kadiealexander this issue is now 4 weeks old and preventing us from maintaining WAQ. This should now be your highest priority. Please post below what your plan is to get a PR in review ASAP. Thanks!

**Comment by kadiealexander** [comment]
Not overdue.

**Comment by Julesssss** [comment]
> Note: all e/App bugs should be Daily unless they are on Hold. I am flipping this back to daily as part of the weekly resync chore comb through.

I missed this message. 

This isn't a bug, it's a new feature request so shouldn't be daily. I updated the labels.

**Comment by kadiealexander** [comment]
@Julesssss @eVoloshchak any update on this one?
